plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RewriteLpad.java

```diff
@@ -0,0 +1,97 @@
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.functionName;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.type;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.variable;
+import static io.trino.spi.type.IntegerType.INTEGER;
+
+public class RewriteLpad
+        implements ProjectFunctionRule<JdbcExpression, ParameterizedExpression>
+{
+    private static final Capture<Variable> ARGUMENT1 = newCapture();
+    private static final Capture<Constant> ARGUMENT2 = newCapture();
+    private static final Capture<Constant> ARGUMENT3 = newCapture();
+
+    private static final Pattern<Call> PATTERN = call()
+            .with(functionName().equalTo(new FunctionName("lpad")))
+            .with(type().matching(type -> type instanceof VarcharType))
+            .with(argumentCount().equalTo(3))
+            .with(argument(0).matching(variable().capturedAs(ARGUMENT1).with(type().matching(type -> type instanceof VarcharType))))
+            .with(argument(1).matching(constant().capturedAs(ARGUMENT2).with(type().matching(type -> type instanceof BigintType))))
+            .with(argument(2).matching(constant().capturedAs(ARGUMENT3).with(type().matching(type -> type instanceof VarcharType))));
+
+    @Override
+    public Pattern<? extends ConnectorExpression> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(ConnectorTableHandle handle, ConnectorExpression projectionExpression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        Variable argument1 = captures.get(ARGUMENT1);
+        Constant argument2 = captures.get(ARGUMENT2);
+        Constant argument3 = captures.get(ARGUMENT3);
+
+        JdbcTypeHandle typeHandle = ((JdbcColumnHandle) context.getAssignment(argument1.getName())).getJdbcTypeHandle();
+
+        ParameterizedExpression translatedArgument1 = context.rewriteExpression(argument1).orElseThrow();
+        ParameterizedExpression translatedArgument2 = context.rewriteExpression(argument2).orElseThrow();
+        ParameterizedExpression translatedArgument3 = context.rewriteExpression(argument3).orElseThrow();
+
+        // Pass empty columnSize to match unbounded varchar type created for synthetic column created for lpad expression.
+        JdbcTypeHandle updatedTypeHandle = new JdbcTypeHandle(typeHandle.jdbcType(), typeHandle.jdbcTypeName(), Optional.empty(), typeHandle.decimalDigits(), typeHandle.arrayDimensions(), typeHandle.caseSensitivity());
+
+        QueryParameter queryParameter = translatedArgument2.parameters().getFirst();
+        // Redshift requires integer type for size parameter. With bigint, driver throws `function lpad(character varying, bigint, "unknown") does not exist`
+        QueryParameter updatedQueryParameter = new QueryParameter(INTEGER, queryParameter.getValue());
```

Comment:
Since Trino allows `targetLength` as `Long`. How this will behave when the `queryParameter.getValue()` is greater than `INTEGER.MAX`?